api/python/t4/api.py

```diff
@@ -208,23 +208,188 @@ def list_packages(registry=None):
     Returns:
         A list of strings containing the names of the packages
     """
-    registry = get_package_registry(fix_url(registry) if registry else None) + '/named_packages'
+    class PackageList:
+        """Display wrapper for list_packages"""
+
+        def __init__(self, pkg_info):
+            self.pkg_names = [info[0].replace(':latest', '') for info in pkg_info]
+            self._repr = self.create_str(pkg_info)
+
+        def __repr__(self):
+            return self._repr
+
+        def __iter__(self):
+            return iter(self.pkg_names)
 
-    registry_url = urlparse(registry)
+        def __len__(self):
+            return len(self.pkg_names)
+
+        def __contains__(self, item):
+            return item in self.pkg_names
+
+        @staticmethod
+        def _fmt_str(string, strlen):
```

Comment:
pull the end of line column (three spaces out of here)

Reply:
I'm not sure what you mean by that.

Reply:
Yeah, poorly written. I meant move the three space you are using to divide columns ot of the `fmt_str` funciton, which you've done.